iiblib/iib_build_details_model.py

```diff
@@ -61,11 +61,37 @@ class IIBBuildDetailsModel(object):
 
     _operation_attrs = []
 
+    _accepts = ""
+
     def __init__(self, *args, **kwargs):
         self._data = self._get_args(kwargs)
 
     @classmethod
     def from_dict(cls, data):
+        """
+        Create an object from dictionary
+        """
+
+        if (
+            cls._accepts != data["request_type"]
```

Comment:
Since we went with this approach we can also check refactor this method so it loads necessary data dynamically. So we do not have to take care about it when new classes are created:

```python
  @classmethod
    def _validate_data(cls, data):

        accepted_request_types = [
            sub_cls._accepts for sub_cls in IIBBuildDetailsModel.__subclasses__()
        ]

        if data["request_type"] not in accepted_request_types:
            raise KeyError("Unsupported request type: %s" % data["request_type"])

        if (
            cls._accepts != data["request_type"]
            and cls.__name__ != IIBBuildDetailsModel.__name__
        ):
            raise TypeError(
                "Class %s doesn't accept %s request type",
                (cls.__name__, data["request_type"]),
            )

    @classmethod
    def from_dict(cls, data):
        """
        Create an object from dictionary
        """

        cls._validate_data(data)

        request_type_to_class = {
            sub_cls._accepts: sub_cls
            for sub_cls in IIBBuildDetailsModel.__subclasses__()
        }

        return request_type_to_class[data["request_type"]](**data)

```

Plus also add comments and docstring for parameters, what is returned / raised and so on.

Reply:
Changes were made